### ISSUE TITLE
project needs a rename

### ISSUE THREAD

**Issue by aspiers** [openSUSE/obs-service-tar_scm#158]
As seen in #157 this repo has extended its scope quite a lot and now provides 5 different services, only two of which are related to `tar`.  So I propose we rename the project.  

To be clear, I am *not* proposing renaming the services (which already have good names) - just the project.  I guess this should only require renaming the github repo and then repointing files like https://build.opensuse.org/package/view_file/openSUSE:Tools/obs-service-tar_scm/_service?expand=1 to it, so it should be a small change.

Any thoughts on this and suggestions for new names? /cc @adrianschroeter @M0ses

**Comment by M0ses** [comment]
I like the idea of renaming because the services provided nowadays have nothing to do with the origin once, even I have no suggestions for a good "new" name.

**Comment by M0ses** [comment]
@aspiers @adrianschroeter what do you think about obs-service-scm2obs ?

**Comment by jengelh** [comment]
What is the point of having `obs` twice in the name — just use `obs-service-scm`.

**Comment by aspiers** [comment]
@jengelh That's a good point.

The services are:

- `tar_scm` (obviously) - input: SCM-controlled source trees, output: tar ball
- `obs_scm` - input: SCM-controlled source trees, output: obscpio archive and `<package>.obsinfo`
- `tar` - input: obscpio archive files, output: tar ball
- `snapcraft` - input: YAML file, output: ???
- `appimage` - input: YAML file, output: ???

I'm trying to figure out what they all have in common, but as highlighted in #156 I don't understand enough about snapcraft / appimage yet to suss this out.

**Comment by M0ses** [comment]
@aspiers : snapcraft and appimage are also able to use SCM systems to get the code - I think this is why @adrianschroeter included them into tar_scm - to make usage of the given "infrastructure". Both of them could contain more then one SCM config and have their own format to specify the SCM repo (using yaml).

**Comment by aspiers** [comment]
@M0ses so they get the YAML from SCM, or just other code which the YAML references?  And what is the output of these source services?  Do they build a snap and AppImage app respectively, or is that done during the OBS build phase?  It would be really helpful if there was a more detailed explanation somewhere on how they work - I'm still struggling to understand it from the short descriptions in the README.

**Comment by M0ses** [comment]
> @M0ses so they get the YAML from SCM, or just other code which the YAML references?
> And what is the output of these source services? 

code which is referenced in their YAML files

>Do they build a snap and AppImage app respectively, or is that done during the OBS build phase? 

I think I already answered this question in https://github.com/openSUSE/obs-service-tar_scm/issues/156#issuecomment-329207740

IMHO - the renaming of the project should be a different topic than missing/incomplete documentation, rigth? And there are other issues open for that.

**Comment by aspiers** [comment]
> IMHO - the renaming of the project should be a different topic than missing/incomplete documentation, rigth? And there are other issues open for that.

@M0ses Yes definitely, but until I understand how it all works, I will struggle to think of a good new name for the project :)

**Comment by adrianschroeter** [comment]
what about obs-service-scm  since all the services are around some kind of SCM handling, no?

**Comment by aspiers** [comment]
@adrianschroeter the `tar` service doesn't use SCM, or am I misunderstanding?

**Comment by aspiers** [comment]
So now I am wondering why the `tar` service was added to this repo.  The code treats `tar` as if it is an SCM like `git`, but obviously that's not what `tar` is.  However it looks like there is some genuine benefit being derived from having a [`TarSCM.scm.tar` class which inherits from `TarSCM.scm.base`](https://github.com/openSUSE/obs-service-tar_scm/blob/master/TarSCM/scm/tar.py), so this makes me suspect that the scope of the base class has somehow grown beyond just SCMs and needs a rename.  Unfortunately it's difficult for me to know because none of this new code has docstrings.  I am going to submit a new issue about that.

**Comment by aspiers** [comment]
> I am going to submit a new issue about that.

Submitted as #195.

**Comment by aspiers** [comment]
I think I've managed to answer a bunch of my questions now.  My current understanding of the services:

- `tar_scm` (obviously) - input: SCM-controlled source trees, output: tar ball
- `obs_scm` - input: SCM-controlled source trees, output: obscpio archive and `<package>.obsinfo`
- `tar` - input: obscpio archive files, output: tar ball
- `snapcraft` - input: YAML file, output: tar balls
- `appimage` - input: YAML file, output: tar balls

@M0ses @adrianschroeter Is that right?  If so, that means:

- All but one service (`obs_scm`) outputs one or more tarballs
- All but one service (`tar`) retrieves sources via an SCM

In other words, all 5 services have something to do with either SCMs, or tar, and 3 of the 5 use both.  And in reality, `obs_scm` and `tar` are just two halves of the same pipeline (SCM -> obscpio -> tar), so in that light, they are still converting from SCM into tarballs, there's just an extra step in the middle - right?

So my current thinking is that we could keep the name `tar_scm`, but change the interpretation of the name to mean "tar OR scm", or "tarballs built directly or indirectly from SCMs", rather than the old interpretation which was "tarballs immediately built from an SCM" :-)  

Does that make sense?  Everyone please let me know your thoughts on this!

**Comment by M0ses** [comment]
I`m completly fine with your new interpretation of the name and I think it saves a lot of time to keep the old name. As you raised this issue, it`s up to you to close it  :-).

**Comment by aspiers** [comment]
OK great, I think I'll do a final tweak to the README before closing :-)
